ragna/assistants/_demo.py

```diff
@@ -39,16 +35,19 @@ def _markdown_answer(self) -> str:
             """
         ).strip()
 
-    def _default_answer(self, prompt: str, sources: list[Source]) -> str:
+    def _default_answer(self, messages: list[Message]) -> str:
+        prompt = messages[-1].content.strip()
```

Comment:
Either need to trust the order of `messages` or make sure `core.Message` inherits timestamp from `schema.Message`

Reply:
We can trust the order of the messages. When using only the Python API, we are just appending messages to a list. As for restoring from DB, we explicitly sort them

https://github.com/Quansight/ragna/blob/9776ec5b06b573ee0b672e07e23c13cd7e0e5fb3/ragna/deploy/_api/orm.py#L89-L91

Reply:
Great, that makes it easy